scalyr_agent/util.py

```diff
@@ -1257,6 +1257,25 @@ def charge_if_available(self, num_bytes, current_time=None):
 
         @return: True if there are enough room in the rate limit to allow the operation.
         """
+        if self._get_time_to_sleep(num_bytes, current_time) > 0.0:
+            return False
+        else:
+            self.__bucket_contents -= num_bytes
+            return True
+
+    def sleep_and_charge(self, num_bytes, current_time=None):
+        """Sleeps until there are enough bytes available for an operation costing num_bytes, then charges that amount.
+
```

Comment:
Also, I see I didn't call this out originally (that the abstraction is not thread-safe.)  Can you go ahead and add a comment to the class comment about it not being thread safe please?